service/s3/unmarshal_error.go

```diff
@@ -48,6 +53,7 @@ func unmarshalError(r *request.Request) {
 	} else {
 		errCode = resp.Code
 		errMsg = resp.Message
+		err = nil
```

Comment:
yeah... Look at this i'm not sure why the `io.EOF` check is here. I'll dig a bit to see if i can find a reason. I think the EOF condition can be removed.